DEVELOPERS.md

```diff
@@ -11,47 +11,109 @@ The `just` command provides a list of available recipes:
 just list
 ```
 
-Running any of the test/check/fix `just` commands will setup a local environment and
-install dependencies.
+Running any of the `just` commands that need it will set up a local environment and install dependencies.
 
+## Testing
 
-## Test setup
+### Test categories
 
-The test suite includes:
-- integration tests: tests that use Docker (mostly to run a database, but also in some cases to test the data builder's
-  own Docker image)
-- unit tests: tests that don't require Docker (and therefore don't query an actual database)
+Tests are divided into the following categories.
 
-We run the database for the integration tests in a Docker container. Each run of the tests starts the
-database if it's not already running _and then leaves it running_ at the end to speed up future runs. (Each test cleans
-out the schema to avoid pollution.)
+<dl>
+   <dt>unit</dt><dd>fast tests of small code units</dd>
+   <dt>spec</dt><dd>tests generated from the ehrQL spec</dd>
+   <dt>acceptance</dt><dd>tests which check compatibility with real studies</dd>
+   <dt>integration</dt><dd>tests of detailed code logic that require a database</dd>
```

Comment:
Where do tests which exercise the CLI entrypoint (but not via Docker) belong? I think I would have described them as integration tests, but they don't always involve a database (at least the ones I'm writing now don't).

Reply:
Yes. I had completely forgotten about that trivial study test until you asked about this, so it's definitely missing from the description.

Having not yet seen it, I'm mildly against your error path test in there, but I'm happy to wait and see.